outscale/resource_outscale_api_access_rule.go

```diff
@@ -204,7 +214,9 @@ func resourceOutscaleOAPIApiAccessRuleDelete(d *schema.ResourceData, meta interf
 		}
 		return nil
 	})
-
+	if err != nil {
+		return err
+	}
 	return nil
```

Comment:
the if seems useless, you can simply return err, as if error is nil, you don't go in the above condition